app/models/like.rb

```diff
@@ -0,0 +1,12 @@
+class Like < ApplicationRecord
+  belongs_to :post, foreign_key: true, optional: true
+
+  def update_post_likecount
+    post = Post.find(post_id)
+
+    post.likes_count = 0 if post.likes_count.nil?
+
+    post.likes_count += 1
+    post.save
+  end
```

Comment:
- [ OPTIONAL] You could also use the Callback after_save here as proposed for the posts counts and aslo apply good naming convention for your method name.